pkg/lightning/lightning.go

```diff
@@ -271,7 +271,11 @@ func (l *Lightning) run(taskCtx context.Context, taskCfg *config.Config, g glue.
 	if err != nil {
 		return errors.Annotate(err, "parse backend failed")
 	}
-	s, err := storage.Create(ctx, u, true)
+	s, err := storage.New(ctx, u, &storage.ExternalStorageOptions{
+		// we skip check path in favor of delaying the error to when we actually access the file.
+		// on S3, performing "check path" requires the additional "s3:ListBucket" permission.
+		SkipCheckPath: true,
```

Comment:
What about BR? Does it need the change too?

Reply:
See #898 for details. Yes BR also needs to be fixed, but not via disabling SkipCheckPath everywhere.